webextension/background.js

```diff
@@ -129,8 +127,76 @@ const Config = (function() {
       }
     }
 
-    load() {
+    maybeSendTelemetry(message) {
+      browser.study.sendTelemetry(message);
+    }
+
+    get shieldStudySetup() {
+      const pref = `extensions.${browser.runtime.id.split("@")[0]}.variation`;
```

Comment:
I think this may blow up if the `browser.runtime.id` doesn't have at least one `@`, but maybe that's OK if our id is valid.

Reply:
I *think* it will be fine as long (as it's a string), since the two variants I would expect here are an email address or a random GUID if the addon doesn't provide one.

Reply:
The alternative is to just hard code the name, `experiment.webcompat-blipz-experiment.variation` -- I don't think we have any reason to believe this will be dynamic.

Reply:
I guess either way it wouldn't matter, if it's expected to remain static. So I'll let you two decide :)

Reply:
Assuming gecko.id isn't changing from manifest.json, it's fine to leave for now.

Reply:
we use this in the utils:
```
    function makeWidgetId(id) {
      id = id.toLowerCase();
      // FIXME: This allows for collisions.
      // WebExt hasn't ever had a problem.
      return id.replace(/[^a-z0-9_-]/g, "_");
    }

    const widgetId = makeWidgetId(
      this._extensionManifest.applications.gecko.id,
    );
```